## Exploración de datos

### Índice
1. [Descripción de los datos ](#id1)
2. [Outliers](#id2)
3. [Distribución de los datos](#id3)
4. [Visualización de los datos](#id4)
5. [Conclusión](#id5)

### 1. Descripción de los datos<a name="id1"></a>
Se describen la cantidad de registros y columnas con sus tipos de datos, ocupación en memoria y una estadística descriptiva de los mismos: cúal es el promedio, el desvío estandar, los cuartiles, todos medidas de tendencia central.

In [1]:
# se importan las librerias necesarias
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np


In [2]:
# Se importa el dataset de trabajo
df = pd.read_csv('./dataset/movies_limpio.csv',  delimiter=',')

C:\Users\Area-Desa15\AppData\Local\Temp\ipykernel_14760\1844338907.py:2: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('./dataset/movies_limpio.csv',  delimiter=',')


In [3]:
df.head() # Se observan los primeros registros

budget                                             genres     id  \
0  30000000.0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...    862   
1  65000000.0  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   8844   
2         0.0  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...  15602   
3  16000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  31357   
4         0.0                     [{'id': 35, 'name': 'Comedy'}]  11862   

  original_language                                           overview  \
0                en  Led by Woody, Andy's toys live happily in his ...   
1                en  When siblings Judy and Peter discover an encha...   
2                en  A family wedding reignites the ancient feud be...   
3                en  Cheated on, mistreated and stepped on, the wom...   
4                en  Just when George Banks has recovered from his ...   

  popularity release_date      revenue  runtime  \
0  21.946943   1995-10-30  373554033.0     81.0   
1  17.015539   1995-12-15  262797249.0    104.0   
2    11.7129   1995-12-22          0.0    101.0   
3   3.859495   1995-12-22   81452156.0    127.0   
4   8.387519   1995-02-10   76578911.0    106.0   

                                    spoken_languages    status  \
0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
1  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released   
2           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
3           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
4           [{'iso_639_1': 'en', 'name': 'English'}]  Released   

                                             tagline  \
0                                                NaN   
1          Roll the dice and unleash the excitement!   
2  Still Yelling. Still Fighting. Still Ready for...   
3  Friends are the people who let you be yourself...   
4  Just When His World Is Back To Normal... He's ...   

                         title  vote_average  vote_count  \
0                    Toy Story           7.7      5415.0   
1                      Jumanji           6.9      2413.0   
2             Grumpier Old Men           6.5        92.0   
3            Waiting to Exhale           6.1        34.0   
4  Father of the Bride Part II           5.7       173.0   

                       franquicia  \
0            Toy Story Collection   
1                             NaN   
2       Grumpy Old Men Collection   
3                             NaN   
4  Father of the Bride Collection   

                                         productoras  \
0                            Pixar Animation Studios   
1  TriStar Pictures, Teitler Film, Interscope Com...   
2                       Warner Bros., Lancaster Gate   
3             Twentieth Century Fox Film Corporation   
4         Sandollar Productions, Touchstone Pictures   

                     paises  release_year    retorno  
0  United States of America        1995.0  12.451801  
1  United States of America        1995.0   4.043035  
2  United States of America        1995.0   0.000000  
3  United States of America        1995.0   5.090760  
4  United States of America        1995.0   0.000000

In [5]:
df.columns # se listan las columnas

Index(['budget', 'genres', 'id', 'original_language', 'overview', 'popularity',
       'release_date', 'revenue', 'runtime', 'spoken_languages', 'status',
       'tagline', 'title', 'vote_average', 'vote_count', 'franquicia',
       'productoras', 'paises', 'release_year', 'retorno'],
      dtype='object')

In [6]:
print('Cantidad de Filas y Columnas',df.shape)

Cantidad de Filas y Columnas (45379, 20)


In [7]:
df.describe(include="all")  # Estadistica descriptiva
# Esto es un resumen del conjunto de datos para todas las columnas numéricas, excepto los valores NaN

budget                         genres      id original_language  \
count   4.537900e+04                          45379   45379             45368   
unique           NaN                           4068   45349                92   
top              NaN  [{'id': 18, 'name': 'Drama'}]  141971                en   
freq             NaN                           4998       3             32202   
mean    4.232325e+06                            NaN     NaN               NaN   
std     1.743932e+07                            NaN     NaN               NaN   
min     0.000000e+00                            NaN     NaN               NaN   
25%     0.000000e+00                            NaN     NaN               NaN   
50%     0.000000e+00                            NaN     NaN               NaN   
75%     0.000000e+00                            NaN     NaN               NaN   
max     3.800000e+08                            NaN     NaN               NaN   

                  overview popularity release_date       revenue  \
count                44438      45377        45376  4.537900e+04   
unique               44233      44142        17333           NaN   
top     No overview found.      1e-06   2008-01-01           NaN   
freq                   133         30          136           NaN   
mean                   NaN        NaN          NaN  1.122936e+07   
std                    NaN        NaN          NaN  6.438789e+07   
min                    NaN        NaN          NaN  0.000000e+00   
25%                    NaN        NaN          NaN  0.000000e+00   
50%                    NaN        NaN          NaN  0.000000e+00   
75%                    NaN        NaN          NaN  0.000000e+00   
max                    NaN        NaN          NaN  2.787965e+09   

             runtime                          spoken_languages    status  \
count   45130.000000                                     45376     45296   
unique           NaN                                      1931         6   
top              NaN  [{'iso_639_1': 'en', 'name': 'English'}]  Released   
freq             NaN                                     22380     44936   
mean       94.181675                                       NaN       NaN   
std        38.341059                                       NaN       NaN   
min         0.000000                                       NaN       NaN   
25%        85.000000                                       NaN       NaN   
50%        95.000000                                       NaN       NaN   
75%       107.000000                                       NaN       NaN   
max      1256.000000                                       NaN       NaN   

                       tagline       title  vote_average    vote_count  \
count                    20398       45376  45376.000000  45376.000000   
unique                   20269       42196           NaN           NaN   
top     Based on a true story.  Cinderella           NaN           NaN   
freq                         7          11           NaN           NaN   
mean                       NaN         NaN      5.624070    110.096439   
std                        NaN         NaN      1.915423    491.742894   
min                        NaN         NaN      0.000000      0.000000   
25%                        NaN         NaN      5.000000      3.000000   
50%                        NaN         NaN      6.000000     10.000000   
75%                        NaN         NaN      6.800000     34.000000   
max                        NaN         NaN     10.000000  14075.000000   

             franquicia                productoras                    paises  \
count              4488                      33096                     39160   
unique             1695                      22246                      2383   
top     The Bowery Boys  Metro-Goldwyn-Mayer (MGM)  United States of America   
freq                 29                        742                     17846   
mean                Na

In [8]:
df.describe() # describe reducido a las columnas numericas

budget       revenue       runtime  vote_average    vote_count  \
count  4.537900e+04  4.537900e+04  45130.000000  45376.000000  45376.000000   
mean   4.232325e+06  1.122936e+07     94.181675      5.624070    110.096439   
std    1.743932e+07  6.438789e+07     38.341059      1.915423    491.742894   
min    0.000000e+00  0.000000e+00      0.000000      0.000000      0.000000   
25%    0.000000e+00  0.000000e+00     85.000000      5.000000      3.000000   
50%    0.000000e+00  0.000000e+00     95.000000      6.000000     10.000000   
75%    0.000000e+00  0.000000e+00    107.000000      6.800000     34.000000   
max    3.800000e+08  2.787965e+09   1256.000000     10.000000  14075.000000   

       release_year       retorno  
count  45376.000000  4.537900e+04  
mean    1991.881193  6.599991e+02  
std       24.055360  7.469083e+04  
min     1874.000000  0.000000e+00  
25%     1978.000000  0.000000e+00  
50%     2001.000000  0.000000e+00  
75%     2010.000000  0.000000e+00  
max     2020.000000  1.239638e+07

Con los análisis anteriores se puede observar que las peliculas registras datan de los años 1874 al 2020. Entre ellas hay filmaciones con presupuesto y ganancia en cero. 
También se observa que hay peliculas que no tienen valoración registrada, en cuanto a puntuación por polularidad TMDB (TheMoviesDataBase), reseña y votación.

Se muestra cúales son esos registros:

In [9]:
# Se observa por columna: la cantidad de registro, el tipo de dato y si tiene valores nulos o no. 
# También se observa el uso de memoria.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45379 entries, 0 to 45378
Data columns (total 20 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   budget             45379 non-null  float64
 1   genres             45379 non-null  object 
 2   id                 45379 non-null  object 
 3   original_language  45368 non-null  object 
 4   overview           44438 non-null  object 
 5   popularity         45377 non-null  object 
 6   release_date       45376 non-null  object 
 7   revenue            45379 non-null  float64
 8   runtime            45130 non-null  float64
 9   spoken_languages   45376 non-null  object 
 10  status             45296 non-null  object 
 11  tagline            20398 non-null  object 
 12  title              45376 non-null  object 
 13  vote_average       45376 non-null  float64
 14  vote_count         45376 non-null  float64
 15  franquicia         4488 non-null   object 
 16  productoras        330

In [10]:
df.isnull().sum() # se enlista los nulos por columna, pero no se los imputa.

budget                   0
genres                   0
id                       0
original_language       11
overview               941
popularity               2
release_date             3
revenue                  0
runtime                249
spoken_languages         3
status                  83
tagline              24981
title                    3
vote_average             3
vote_count               3
franquicia           40891
productoras          12283
paises                6219
release_year             3
retorno                  0
dtype: int64

In [12]:
# se analiza si hay duplicados:
df.duplicated().sum()

13

In [14]:
print(df[df.duplicated(keep=False)]) # se visualizan los registros duplicados

         budget                                             genres      id  \
676         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...  105045   
1463        0.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...  105045   
7340        0.0  [{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...    5511   
8063        0.0  [{'id': 12, 'name': 'Adventure'}, {'id': 16, '...   23305   
9160        0.0  [{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...    5511   
9322        0.0  [{'id': 12, 'name': 'Adventure'}, {'id': 16, '...   23305   
13256       0.0  [{'id': 53, 'name': 'Thriller'}, {'id': 9648, ...  141971   
13370       0.0  [{'id': 53, 'name': 'Thriller'}, {'id': 9648, ...  141971   
13598       0.0  [{'id': 18, 'name': 'Drama'}, {'id': 35, 'name...  168538   
13994       0.0  [{'id': 28, 'name': 'Action'}, {'id': 27, 'nam...   18440   
14006       0.0  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   11115   
15695       0.0  [{'id': 18, 'name': 'Drama'}, {'id': 10769, 'n.

In [16]:
df = df.drop_duplicates()
print('Cantidad de registros sin duplicados',df.shape[0])

Cantidad de registros sin duplicados 45366


### 2. Ortliers<a name="id2"></a>
Detección de valores atípicos.